workflows/argo/prognostic-run.yaml

```diff
@@ -22,12 +22,25 @@ spec:
           - {name: initial-condition}
           - {name: config}
           - {name: reference-restarts}
+          - {name: tag}
           - {name: flags, value: ""}
-          - {name: output}
+          - {name: bucket, value: "vcm-ml-experiments"}
+          - {name: project, value: "default"}
+          - {name: step, value: "prognostic_run"}
```

Comment:
is there a particular reason that the `step` name is a parameter for the prognostic run? I thought we were just gonna hard-code it for all the workflow steps.

Reply:
Yeah, because the `prognostic-run` workflow is also the N2F and N2O workflow, and presumably we want to give those steps different labels than the prognostic run.

Reply:
Do we want to give them different labels though? I think that's an important point to discuss. At the end of the day, they're all FV3GFS runs and all being executed with the same argo workflow. It does mean that we could not have multiple runs under the same tag (which is currently done for the e2e test, for example).

Reply:
I agree with @brianhenn. To me the relevant point is that nudge to fine runs contain the same outputs as other nudge to fine runs (and have their own example folder in vcm-workflow-control). The process that generated the data is less important to capture in the path name. As a user, I want to do queries like "What are all nudge to fine runs that can be loaded with `open_nudge_to_fine`.

Reply:
Fair point about the baseline case, I think I've been inconsistent between 'prognostic_run' and 'baseline_run' and we don't have a vcm-workflow-control example to guide us.

Reply:
It's a good point. It's hard to capture the polymorphic nature of data produced by the prognostic run in a URL. In theory the outputs available in a run directory can be queried by looking up the fv3config.yml or listing the zarrs available, all segmented runs fit this basic format. Should we call it "segmented-run"?

To enable more efficient querying of arbitrary metadata (e.g. all C48 prog runs that had nudging on for X Y and Z), we could maintain a separate index.

I'm with @oliverwm1  now.

Reply:
How does an e2e type experiment get written out under this approach?

Reply:
The e2e test is unique in that it's our only argo workflow that submits multiple prognostic runs. Under this framework, I guess you'd have to have a different `tag` for the nudge-to-fine run as you do for the `train-evaluate...` workflow.

Reply:
right..some workflows run a single step multiple times. One could see the same problem with a training an ensemble of nets. Is it important to provide this grouping in the experiment tag? If not then we could it append info to the tag? 

```
gs://bucket/default/2021-01-01/segmented-run/{tag}-nudge-to-fine/
```

This would still allow fast querying by the prefix `gs://bucket/default/2021-01-01/*/{tag}`

Reply:
I think I'll leave it up to @brianhenn for what to do on this. It's more of a breaking change from our previous approach. I think the prognostic run vs nudged approach is meaningful enough to warrant being in separate directories.

Reply:
Honestly, my preferred naming nomenclature would be

```
gs://{bucket}/{step}/d1f6fa7e-a9f4-11eb-b831-3ff9a969d9d6
```

with a `metadata.yaml` in each folder.

And we could write a small library to do things like list in order of time, experiment, config values. etc.

Reply:
Building the query tools seems worthwhile but also pretty sure there's value to the group in terms of human-readable dates and tags :)

What about calling all fv3gfs runs just that -- `fv3gfs_run`? segmented doesn't seem like the salient point about them, and the nudged runs aren't really prognostic. And then yeah just relying on the tags to differentiate them.

So for an e2e type experiment we'd have:
`gs://{bucket}/{project}/2021-04-30/{tag}/fv3gfs_run` where tag is something like `n2f-3km-nudge-to-fine`
and:
`gs://bucket/project/2021-04-30/{tag}/fv3gfs_run`
`gs://bucket/project/2021-04-30/{tag}/trained_model`
`gs://bucket/project/2021-04-30/{tag}/offline_diags`
`gs://bucket/project/2021-04-30/{tag}/fv3gfs_run`
...
where tag is something like `n2f-3km-train-diags-prog` .

Model ensembles would have to be handled with tags also, e.g.:
`gs://bucket/project/2021-04-30/n2f-3km-nn-random-seed1/trained_model`
`gs://bucket/project/2021-04-30/n2f-3km-nn-random-seed2/trained_model`
...
But that seems to be basically what Anna has been doing anyway.

Any objections?

Reply:
If we chose one name for all of the runs I think my preference order is
1a) fv3gfs_run
1b) prognostic_run (nudging runs are kind of prognostic I guess and we already use this)
2) segmented_run

Reply:
I agree with your suggestions excepting this one:

> 'segmented' doesn't seem like the salient point about them, and the nudged runs aren't really prognostic. 

To me, segmented runs are special compared to the output of `runfv3 run-native` for a few reasons:
1. RESTART and log files are in the `artifacts` sub directory
2. The metadata on the .zarr files is consolidated. They contain some metadata at the root level that is shared by all the runs in `artifacts`.

Reply:
but I am fine with any of your proposed names.

Reply:
I would be +1 on `fv3gfs_run`. Although I agree the segmented runs are special because of their output format, it's not possible for the argo workflows to output data in the `run-native` format anyway, so not sure the distinction is that useful. Anyway I'm also fine with any of the proposed names.

Reply:
Implemented using `fv3gfs_run`.